csrc/ir/builder.cpp

```diff
@@ -391,7 +401,7 @@ Val* SimplifyingIrBuilder::ceilDivExpr(Val* lhs, Val* rhs) {
 }
 
 Val* SimplifyingIrBuilder::modExpr(Val* lhs, Val* rhs) {
-  if (rhs->isOneInt()) {
+  if (rhs->isOneInt() || lhs->isZeroInt() || lhs->sameAs(rhs)) {
```

Comment:
Should we `NVFUSER_ERROR` that both `lhs` and `rhs` are integral types?

Reply:
I think I can just guard the last condition here with `lhs->isIntegralScalar()`.

Reply:
I'll do that.